### PR TITLE
#34 add server settings UI

### PR DESCRIPTION
**Pull request by wlthomson**
Adds `SyncSettingsScreen` for basic sync configuration/credentials.

### CODE REVIEW

File: src/features/Sync/SyncSlice.ts [hunk 1]
```diff
@@ -18,6 +91,51 @@ const { actions: SyncAction, reducer: SyncReducer } = createSlice({
...
+}
+
+const SyncSelector = {
+    getUrl,
+    getUsername,
+    getPassword
+};
+
+function* updateUrl({ payload: { url }}: UpdateUrlAction): SagaIterator {
+    const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
```
Comment (joshxg) [inline src/features/Sync/SyncSlice.ts:120]: assuming persistance storage is later?

We will need urls for each of the endpoints

File: src/features/Sync/SyncSlice.ts [hunk 2]
```diff
@@ -18,6 +91,51 @@ const { actions: SyncAction, reducer: SyncReducer } = createSlice({
...
+    const syncOutManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_OUT_MANAGER);
+    syncOutManager.setUrl(url);
```
Comment (joshxg) [inline src/features/Sync/SyncSlice.ts:122]: personally think passing the manager these values is easier then keeping 3 sources of the values - two in memory and one in persistant storage

File: src/ui/screens/settings/MainSettingsScreen.tsx [hunk 3]
```diff
@@ -38,7 +38,12 @@ export const MainSettingsScreen: FC<MainSettingsScreenProps> = ({ navigation })
           onPress={() => navigation.navigate(SETTINGS_STACK.TEMPERATURE_BREACH)}
           RightComponent={ChevronIcon}
         />
-
+        <SettingsItem
+          label="Sync settings"
```
Comment (joshxg) [inline src/ui/screens/settings/MainSettingsScreen.tsx:42]: can we use the translation mechanism? When the job to translate finally comes would be good to keep it to a minimum (I know there are lots missing!)

**Review by joshxg — APPROVED** [review]
👍

File: src/features/Sync/SyncSlice.ts (hunk 2 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:122]: Ah, I was thinking about separation of front-end (display values in UI) from back-end (use values for syncing) state (but I did also have a concern about maintaining multiple sources-of-truth). For displaying the values in the UI, just get them from the manager?

File: src/ui/screens/settings/MainSettingsScreen.tsx (hunk 3 above)
Comment (wlthomson) [reply src/ui/screens/settings/MainSettingsScreen.tsx:42]: 👍 .

File: src/features/Sync/SyncSlice.ts (hunk 1 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:120]: Yup. Ah, ok. I wasn't sure if the endpoints were standardised!

Comment (joshxg) [reply src/features/Sync/SyncSlice.ts:120]: just as a standalone app integrating with other systems they will likely define their own I guess 🤷

File: src/features/Sync/SyncSlice.ts (hunk 2 above)
Comment (joshxg) [reply src/features/Sync/SyncSlice.ts:122]: I wouldv'e thought read from the store for ui and potentially db for backend

File: src/ui/screens/settings/MainSettingsScreen.tsx (hunk 3 above)
Comment (wlthomson) [reply src/ui/screens/settings/MainSettingsScreen.tsx:42]: Resolving as issue created (#61).

File: src/features/Sync/SyncSlice.ts (hunk 2 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:122]: Resolved by https://github.com/openmsupply/msupply-cold-chain/pull/54.

File: src/features/Sync/SyncSlice.ts (hunk 1 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:120]: Resolved by https://github.com/openmsupply/msupply-cold-chain/pull/54.

**Pull request merged**